src/com/linkedin/parseq/internal/Continuations.java

```diff
@@ -16,8 +16,9 @@
  * such that {@code actionX} can throw exception or recursively call {@code method}
  * multiple times, without worry about stack overflow.
  * <p>
- * The guarantee is that actions are called in the same order than recursive approch
- * would have called them.
+ * The guarantee is that actions are called in the same order than recursive approach
+ * would have called them. To put it in another way, this class guarantees to traversal the
+ * execution tree in post-order.
```

Comment:
Yeah, this class does not claim to be general purpose replacement of recursion - see description in class Javadoc.
The recursivePostOrder in TestContinuations refers to the test case, not the implementation of Continuation class.
The Continuation class implementation is using pre-order traversal.